optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/util/ConcurrentCache.java

```diff
@@ -0,0 +1,31 @@
+package org.optaplanner.core.impl.domain.solution.util;
```

Comment:
Can you move it to the package `org.optaplanner.core.impl.domain.common`? I am not a big fan of `util` packages, as they are a one-name-fits-all package that attrack bloat.